proxy/envoy/config.go

```diff
@@ -354,10 +354,13 @@ func buildHTTPListener(mesh *proxyconfig.MeshConfig, node proxy.Node, instances
 	}
 }
 
-func applyInboundAuth(listener *Listener, mesh *proxyconfig.MeshConfig) {
-	switch mesh.AuthPolicy {
-	case proxyconfig.MeshConfig_NONE:
-	case proxyconfig.MeshConfig_MUTUAL_TLS:
+func shouldApplyAuth(mesh *proxyconfig.MeshConfig, security_option model.SecurityOption) bool {
+	return security_option == model.SecurityEnable ||
+		(mesh.AuthPolicy == proxyconfig.MeshConfig_MUTUAL_TLS && security_option != model.SecurityDisable)
+}
+
+func mayApplyInboundAuth(listener *Listener, mesh *proxyconfig.MeshConfig, security_option model.SecurityOption) {
+	if shouldApplyAuth(mesh, security_option) {
 		listener.SSLContext = buildListenerSSLContext(proxy.AuthCertsPath)
```

Comment:
pass the endpoint to the buildListenerSSLContext. and in that function, if endpoint.Port has secure==false, just return nil. If you are entering buildListenerSSLContext, then you know that global auth was enabled.